instrumentation.go

```diff
@@ -312,21 +350,18 @@ func InstrumentAndLoadCollection(
 				)
 			}
 
-			// Move the symbol from head of the original code to the instrumented block so jumps and function calls
+			// Move the metadata from head of the original code to the instrumented block so jumps and function calls
 			// enter at the instrumented code first.
-			newProgram = append(newProgram, instr[0].WithSymbol(block.Block[0].Symbol()))
+			// source := asm.Comment(fmt.Sprintf("CoverBee[%d]++", blockID))
+			newProgram = append(newProgram, instr[0].WithMetadata(block.Block[0].Metadata))
```

Comment:
Curious, why not keep the copy + clear close together?
```suggestion
			newProgram = append(newProgram, instr[0].WithMetadata(block.Block[0].Metadata))
			block.Block[0] = btf.WithFuncMetadata(block.Block[0].WithSymbol(""), nil)
```
Then you also wouldn't have to check i == 0 on every iteration below.